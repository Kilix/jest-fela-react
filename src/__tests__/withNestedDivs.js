import React from 'react';
import { createComponent, Provider } from 'react-fela';
import { createRenderer } from 'fela';
import renderer from 'react-test-renderer';
import monolithic from 'fela-monolithic';

import createSerializer from '../serializer';

const Text = createComponent(() => ({ color: 'red', fontSize: '16px' }));

const felaRenderer = createRenderer({ enhancers: [monolithic()] });
expect.addSnapshotSerializer(createSerializer(felaRenderer));

it('with-nested-divs', () => {
    const tree = renderer
        .create(
            <Provider renderer={felaRenderer}>
                <div>
                    <Text>Hello World</Text>
                </div>
            </Provider>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
