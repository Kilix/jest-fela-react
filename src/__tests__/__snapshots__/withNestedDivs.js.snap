// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with-nested-divs 1`] = `
.f16lnj6g {
  color: red;
  font-size: 16px;
}

<div>
  <div
    className="f16lnj6g"
  >
    Hello World
  </div>
</div>
`;
